pyoanda/order.py

```diff
@@ -55,9 +56,12 @@ def check(self):
                 type(self.expiry), datetime)
             raise TypeError(msg)
 
-        if hasattr(self, "price") and not isinstance(self.price, (int, float)):
-            msg = "Expiry must be int or float, '{0}' found".format(
-                type(self.price))
-            raise TypeError(msg)
+        if hasattr(self, "price"):
+            try:
+                Decimal(self.price)                
```

Comment:
Its ok, this point it's just a checked and Decimal is wider than float.
I dont thing  is a big penalty for performance, do you?